src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +88,163 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        this.removeObsoleteData(this.schema, schema, this.data);
+        this.loadAndMergeOriginData(this.schema, schema).then(action(()=> {
```

Comment:
i think this would be a bit easier to understand if we first load the data and then manipulate the data. but maybe i am missing some reason why that not possible. i would expect the flow to be something like this:

```js
requestData().then((data) => {
    const mergedData = this.mergeData(oldSchema, newSchema, this.data, data);
    this.setMultiple(mergedData);

    this.schema = schema;
    this.removeUnknownSchemaProperties();
    this.addMissingSchemaProperties();
    this.validate();
    this.setSchemaLoading(false);
)
```

Reply:
For this flow we would have to load the server data on each template switch, not only when going back to the `originTemplate`. I don't see any benefits when loading the `originData` on each template switch. Most likely the data is not compatible with the schema anyway.